modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/collateral/daemon/CollateralDaemon.scala

```diff
@@ -0,0 +1,43 @@
+package org.tessellation.sdk.infrastructure.collateral.daemon
+
+import cats.Applicative
+import cats.effect.{Async, Spawn}
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+
+import org.tessellation.schema.peer
+import org.tessellation.sdk.domain.Daemon
+import org.tessellation.sdk.domain.cluster.storage.ClusterStorage
+import org.tessellation.sdk.domain.collateral.{Collateral, LatestBalances}
+
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+
+trait CollateralDaemon[F[_]] extends Daemon[F] {}
+
+object CollateralDaemon {
+
+  def make[F[_]: Async](
+    collateral: Collateral[F],
+    latestBalances: LatestBalances[F],
+    clusterStorage: ClusterStorage[F]
+  ) = new CollateralDaemon[F] {
+    private val logger = Slf4jLogger.getLogger[F]
+
+    def start: F[Unit] =
+      Spawn[F].start(updatePeersInCluster).void
+
+    private def updatePeersInCluster: F[Unit] =
+      latestBalances.getLatestBalancesStream
+        .evalMap(_ => updateCluster())
+        .compile
+        .drain
+
+    private def updateCluster() =
+      clusterStorage.removePeers(id => collateral.hasCollateral(id).map(!_).flatTap(logRemovingPeer(id)))
```

Comment:
and also, it should be:
```scala
def removePeer(peer: PeerId): F[Unit] = removePeers(Set(peer))
def removePeers(peers: Set[PeerId]: F[Unit] = ??? implementation here
```
instead of
```scala
def removePeer(peer: PeerId): F[Unit] = ??? implementation here
def removePeers(peers: Set[PeerId]: F[Unit] = peers.toList.traverse(removePeer)
```

Reply:
Thanks! Changed.

Reply:
removing a list of peers is faster than iterating one by one